src/Microsoft.AspNet.Mvc.Core/HttpResponseStreamWriter.cs

```diff
@@ -282,7 +282,7 @@ public override Task FlushAsync()
                 throw new ObjectDisposedException("stream");
             }
 
-            return FlushInternalAsync();
+            return FlushInternalAsync(flushEncoder: true);
```

Comment:
Why `true` in either `Flush()` or `FlushAsync()`? This writer can certainly be used after either of those calls.